redact_url.go

```diff
@@ -0,0 +1,46 @@
+package retryablehttp
+
+import (
+	"log/slog"
+	"net/url"
+)
+
+type redactedURL struct{ url *url.URL }
+
+func (r redactedURL) LogValue() slog.Value {
+	return slog.StringValue(r.String())
+}
+
+func (r *redactedURL) String() string {
+	return redactURL(r.url)
+}
+
+var sensitiveParams = []string{
+	"X-Amz-Signature",
+	"X-Amz-Security-Token",
+	"X-Amz-Credential",
+}
+
+func redactURL(u *url.URL) string {
+	if u == nil {
+		return ""
+	}
+
+	ru := *u
+
+	// Taken from url.URL#Redacted() which was introduced in go 1.15.
+	// We can switch to using it directly if we'll bump the minimum required go version.
+	if _, has := ru.User.Password(); has {
+		ru.User = url.UserPassword(ru.User.Username(), "xxxxx")
+	}
+
+	params := ru.Query()
+	for _, param := range sensitiveParams {
```

Comment:
Align this to `wiz/commonlib/anonymize/anonymize.go#RedactURL`, where we omit the q params entirely; if you'd like back compat, you can add an opt-out option for keeping it in the log